streams/dshr_strdata_mod.F90

```diff
@@ -853,18 +1000,32 @@ subroutine shr_strdata_advance(sdat, ymd, tod, logunit, istr, timers, rc)
              if (debug > 0 .and. sdat%masterproc) then
                 write(sdat%logunit,F01) trim(subname),' interp = ',ns,flb,fub
              endif
-
              do nf = 1,size(sdat%pstrm(ns)%fldlist_model)
-                call dshr_fldbun_getfldptr(sdat%pstrm(ns)%fldbun_model, sdat%pstrm(ns)%fldlist_model(nf), dataptr, rc=rc)
-                if (ChkErr(rc,__LINE__,u_FILE_u)) return
-                call dshr_fldbun_getfldptr(sdat%pstrm(ns)%fldbun_data(sdat%pstrm(ns)%stream_lb), &
-                     sdat%pstrm(ns)%fldlist_model(nf), dataptr_lb, rc=rc)
-                if (ChkErr(rc,__LINE__,u_FILE_u)) return
-
-                call dshr_fldbun_getfldptr(sdat%pstrm(ns)%fldbun_data(sdat%pstrm(ns)%stream_ub), &
-                     sdat%pstrm(ns)%fldlist_model(nf), dataptr_ub, rc=rc)
-                if (ChkErr(rc,__LINE__,u_FILE_u)) return
-                dataptr(:) = dataptr_lb(:) * flb + dataptr_ub(:) * fub
+                if (sdat%pstrm(ns)%stream_nlev > 1) then
+                   call dshr_fldbun_getfldptr(sdat%pstrm(ns)%fldbun_model, &
+                        sdat%pstrm(ns)%fldlist_model(nf), fldptr2=dataptr2d, rc=rc)
+                   if (ChkErr(rc,__LINE__,u_FILE_u)) return
+                   call dshr_fldbun_getfldptr(sdat%pstrm(ns)%fldbun_data(sdat%pstrm(ns)%stream_lb), &
+                        sdat%pstrm(ns)%fldlist_model(nf), fldptr2=dataptr2d_lb, rc=rc)
+                   if (ChkErr(rc,__LINE__,u_FILE_u)) return
+                   call dshr_fldbun_getfldptr(sdat%pstrm(ns)%fldbun_data(sdat%pstrm(ns)%stream_ub), &
+                        sdat%pstrm(ns)%fldlist_model(nf), fldptr2=dataptr2d_ub, rc=rc)
+                   if (ChkErr(rc,__LINE__,u_FILE_u)) return
+                   do l = 1,sdat%pstrm(ns)%stream_nlev
```

Comment:
I don't really like 'l' as a variable - it's too close to '1'.

Reply:
I just changed it to lev.